dynawo/sources/Solvers/AlgebraicSolvers/DYNSolverKINAlgRestoration.cpp

```diff
@@ -272,25 +260,15 @@ SolverKINAlgRestoration::evalJ_KIN(N_Vector yy, N_Vector /*rr*/,
 }
 
 int
-SolverKINAlgRestoration::evalJPrim_KIN(N_Vector yy, N_Vector /*rr*/,
+SolverKINAlgRestoration::evalJPrim_KIN(N_Vector /*yy*/, N_Vector /*rr*/,
         SUNMatrix JJ, void* data, N_Vector /*tmp1*/, N_Vector /*tmp2*/) {
   SolverKINAlgRestoration* solv = reinterpret_cast<SolverKINAlgRestoration*> (data);
   shared_ptr<Model> model = solv->getModel();
 
-  double* iyy = NV_DATA_S(yy);
```

Comment:
We assume the differential variables did not change?

Reply:
Normally speaking they have already been copied to the model by the last evalF from the last time step for the first J evaluation. For further evaluations, it would have been copied by the KINAlgRestoration evalF.

Nevertheless, if we have a doubt, we can remove this modification.

Reply:
No it's OK, I guess that you took a close look to results to validate this so it should be OK :)